src/pydvl/influence/general.py

```diff
@@ -125,61 +128,63 @@ def _compute_influences_pert(
     to section 2.2 of :footcite:t:`koh_understanding_2017`.
 
     :param model: A model which has to implement the TwiceDifferentiable interface.
-    :param x_train: An array of shape [MxK] containing the features of the
-        input data points.
-    :param y_train: An array of shape [MxL] containing the targets of the
-        input data points.
+    :param input_data: Data loader containing the samples to calculate the
+        influence of.
     :param influence_factors: array containing influence factors
     :param progress: If True, display progress bars.
     :returns: An array of size [NxMxP], where N is number of test points, M
         number of train points, and P the number of features.
     """
     all_pert_influences = []
-    for i in maybe_progress(
-        len(x),
+    for x, y in maybe_progress(
+        input_data,
         progress,
-        desc="Influence Perturbation",
+        desc="Batch Influence Perturbation",
     ):
-        grad_xy, tensor_x = model.grad(x[i : i + 1], y[i])
-        perturbation_influences = mvp(
-            grad_xy,
-            influence_factors,
-            backprop_on=tensor_x,
-        )
-        all_pert_influences.append(perturbation_influences.reshape((-1, *x[i].shape)))
+        for i in range(len(x)):
+            grad_xy, tensor_x = model.grad(x[i : i + 1], y[i], x_requires_grad=True)
+            perturbation_influences = mvp(
+                grad_xy,
+                influence_factors,
+                backprop_on=tensor_x,
+            )
+            all_pert_influences.append(
+                perturbation_influences.reshape((-1, *x[i].shape))
+            )
 
     return stack(all_pert_influences, axis=1)
 
 
 influence_type_registry = {
-    InfluenceType.Up: _compute_influences_up,
-    InfluenceType.Perturbation: _compute_influences_pert,
+    InfluenceType.Up: compute_influences_up,
+    InfluenceType.Perturbation: compute_influences_pert,
 }
 
 
 def compute_influences(
     differentiable_model: TwiceDifferentiable[TensorType, ModelType],
-    x: TensorType,
-    y: TensorType,
-    x_test: TensorType,
-    y_test: TensorType,
+    training_data: DataLoaderType,
+    test_data: Optional[DataLoaderType] = None,
+    input_data: Optional[DataLoaderType] = None,
     progress: bool = False,
     inversion_method: InversionMethod = InversionMethod.Direct,
+    inversion_method_kwargs: Dict[str, Any] = {},
```

Comment:
Same comment as below:

```suggestion
    **kwargs: Any,
```